aquarius/events/events_monitor.py

```diff
@@ -265,17 +265,29 @@ def handle_token_uri_update(self, from_block, to_block):
                 )
 
     def get_last_processed_block(self):
-        block = 0
         try:
+            # Initialize the cached block with 0 if cache is empty.
+            if get_cached_block() is None:
+                update_cached_block(0)
+            # Re-establishing the connection with ES
+            while self._es_instance.es.ping() is False:
+                logging.error("Trying to connect to ES...")
+                time.sleep(5)
             last_block_record = self._es_instance.es.get(
                 index=self._other_db_index, id=self._index_name, doc_type="_doc"
             )["_source"]
-            block = last_block_record["last_block"]
+            # If blocks indexing starts from 0, get the cached last block instead.
+            block = (
+                last_block_record["last_block"]
+                if last_block_record["last_block"] != 0
```

Comment:
are you sure the value will be an int? Could it possibly be "0" instead?

Reply:
I have a test for that as well and it works as an int value